src/qml/qmlwaveformrenderer.cpp

```diff
@@ -0,0 +1,243 @@
+    renderer->setHighVisualGain(m_gainHigh);
+    connect(this,
+            &QmlWaveformRendererRGB::gainAllChanged,
+            renderer,
+            &WaveformRendererRGB::setAllChannelVisualGain);
+    connect(this,
+            &QmlWaveformRendererRGB::gainLowChanged,
+            renderer,
+            &WaveformRendererRGB::setLowVisualGain);
+    connect(this,
+            &QmlWaveformRendererRGB::gainMidChanged,
+            renderer,
+            &WaveformRendererRGB::setMidVisualGain);
+    connect(this,
+            &QmlWaveformRendererRGB::gainHighChanged,
+            renderer,
+            &WaveformRendererRGB::setHighVisualGain);
+    return QmlWaveformRendererFactory::Renderer{renderer, renderer};
+}
+
+QmlWaveformRendererFactory::Renderer QmlWaveformRendererBeat::create(
+        WaveformWidgetRenderer* waveformWidget) const {
+    auto* renderer = new WaveformRenderBeat(
+            waveformWidget, m_position);
+    renderer->setColor(m_color);
+    connect(this, &QmlWaveformRendererBeat::colorChanged, renderer, &WaveformRenderBeat::setColor);
+    return QmlWaveformRendererFactory::Renderer{renderer, renderer};
+}
+
+QmlWaveformRendererFactory::Renderer QmlWaveformRendererMarkRange::create(
+        WaveformWidgetRenderer* waveformWidget) const {
+    auto* renderer = new WaveformRenderMarkRange(
+            waveformWidget);
+
+    for (auto* pMark : m_ranges) {
+        renderer->addRange(WaveformMarkRange(
+                waveformWidget->getGroup(),
+                pMark->color(),
+                pMark->disabledColor(),
+                pMark->opacity(),
+                pMark->disabledOpacity(),
+                pMark->durationTextColor(),
+                pMark->startControl(),
+                pMark->endControl(),
+                pMark->enabledControl(),
+                pMark->visibilityControl(),
+                pMark->durationTextLocation()));
+    }
+    return QmlWaveformRendererFactory::Renderer{renderer, renderer};
+}
+
+#ifdef __STEM__
+QmlWaveformRendererFactory::Renderer QmlWaveformRendererStem::create(
+        WaveformWidgetRenderer* waveformWidget) const {
+    auto* renderer = new WaveformRendererStem(
+            waveformWidget, m_position);
+
+    renderer->setAllChannelVisualGain(m_gainAll);
+    connect(this,
+            &QmlWaveformRendererStem::gainAllChanged,
+            renderer,
+            &WaveformRendererStem::setAllChannelVisualGain);
+    return QmlWaveformRendererFactory::Renderer{renderer, renderer};
+}
+#endif
+
+QmlWaveformRendererFactory::Renderer QmlWaveformRendererMark::create(
+        WaveformWidgetRenderer* waveformWidget) const {
+    VERIFY_OR_DEBUG_ASSERT(!!m_untilMark) {
+        return QmlWaveformRendererFactory::Renderer{};
+    }
+    auto* renderer = new WaveformRenderMark(waveformWidget,
+            ::WaveformRendererAbstract::Play);
+
+    renderer->setFgPlayColor(m_playMarkerColor);
+    renderer->setBgPlayColor(m_playMarkerBackground);
```

Comment:
I'd suggest to name these consistently and expressive:

```suggestion
    renderer->setPlayMarkerForegroundColor(m_playMarkerForegroundColor);
    renderer->setPlayMarkerBackgroundColor(m_playMarkerBackgroundColor);
```

Reply:
This was auto-generated from the members name, but I think we can rename it as well